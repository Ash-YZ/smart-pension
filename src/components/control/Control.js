import React from 'react'
import PropTypes from 'prop-types'
import './styles.scss'
import Score from '../score/Score'
import Switcher from '../switcher/Switcher'

const Control = props => {
  return (
    <div className='control'>
      <Score score={props.score} />
      <Switcher gameMode='people'
        switchModeHandler={props.switchModeHandler}
        gameModes={['people', 'starship']} />
      <button type="button"
        className="ctrl-button btn btn-success w-100"
        onClick={props.newGameHandler}>{props.submitButtonLabel}</button>
    </div>
  )
}

Control.propTypes = {
  score: PropTypes.arrayOf(PropTypes.number),
  newGameHandler: PropTypes.func,
  switchModeHandler: PropTypes.func,
  submitButtonLabel: PropTypes.string
}

export default Control
