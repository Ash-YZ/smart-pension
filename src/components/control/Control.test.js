import React from 'react';
import { shallow } from 'enzyme'
import Control from './Control'

const submitFn = jest.fn()
const changeFn = jest.fn

const testProps = {
  score: [0, 1],
  newGameHandler: submitFn,
  switchModeHandler: changeFn,
  submitButtonLabel: 'Submit'

}

const component = shallow(<Control {...testProps} />)

describe('Control', () => {
  it('should render control correctly ', () => {
    expect(component).toMatchSnapshot()
  })

  it('should trigger submitFn when the submit button is clicked', () => {
    component.find('.ctrl-button').first().simulate('click')
    expect(submitFn).toHaveBeenCalled()
  })
})