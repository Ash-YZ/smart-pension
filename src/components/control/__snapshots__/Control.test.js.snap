// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control should render control correctly  1`] = `
<div
  className="control"
>
  <Score
    score={
      Array [
        0,
        1,
      ]
    }
  />
  <Switcher
    gameMode="people"
    gameModes={
      Array [
        "people",
        "starship",
      ]
    }
    switchModeHandler={[Function]}
  />
  <button
    className="ctrl-button btn btn-success w-100"
    onClick={[MockFunction]}
    type="button"
  >
    Submit
  </button>
</div>
`;
